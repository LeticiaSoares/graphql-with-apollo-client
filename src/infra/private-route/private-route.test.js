import React from 'react'
import { mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import PrivateRoute from './private-route'
import { MemoryRouter } from 'react-router'

describe('<PrivateRoute />', () => {
    const memoryRouter = <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <PrivateRoute component={() => (<div />)} href='#test' />
    </MemoryRouter>
    Object.defineProperty(
        window,
        'localStorage',
        function () {
            let _storage = {}
            return {
                getItem: function _get (_id) {
                    return _storage._id
                },
                setItem: function _set (_id, _value) {
                    console.log('setItem', _value)
                    _storage = {
                        _id: _value
                    }
                }
            }
        }
    )
    it('renders with snapshot', () => {
        const wrapper = mount(memoryRouter)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })
    it('renders with token', () => {
        localStorage.setItem('auth', JSON.stringify({ token: 'token123456' }))
        const wrapper = mount(memoryRouter)
        expect(wrapper.find('div').exists()).toEqual(true)
    })
    it('renders without token and navigation', () => {
        localStorage.setItem('auth', '')
        mount(memoryRouter)
        expect(window.location.href.split('#')[1]).toEqual('test')
    })
})
