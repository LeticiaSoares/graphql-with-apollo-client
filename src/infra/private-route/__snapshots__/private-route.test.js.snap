// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivateRoute /> renders with snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      "/",
    ]
  }
  initialIndex={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "4ow023",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "4ow023",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PrivateRoute
      component={[Function]}
      href="#test"
    />
  </Router>
</MemoryRouter>
`;
